Black formatting is missing for documentation

awav opened the pull request:
Subj.

st-- commented:
Have you checked that all notebooks are still correctly building after these changes? While I trust that the reformatting doesn't affect the python execution, I'm slightly less sure about how it works with jupytext.

awav commented:
> Have you checked that all notebooks are still correctly building after these changes? While I trust that the reformatting doesn't affect the python execution, I'm slightly less sure about how it works with jupytext.

Yep, that looks fine.

awav commented:
> Could you please rebase your branch and cherry-pick the latest three commits? It looks like you were on a stale branch when you created this branch.

Develop is up-to-date on my laptop.

```bash
(tf) artemav:~/code/GPflow/doc/source/notebooks (develop)
→ git pull
Already up to date.
```

st-- commented:
Not convinced by the double blank lines in notebook code cells, but I guess I can live with that...

st-- commented:
It's also missing for `setup.py` if you want to be comprehensive ?

awav commented on doc/source/notebooks/advanced/kernels.pct.py:
```diff
@@ -200,16 +208,16 @@ def plotkernelfunction(k, ax, xmin=-3, xmax=3, other=0):
+axes[1].imshow(Z[0, :].reshape(n_grid, n_grid), extent=[-10, 10, -10, 10])
+axes[1].set_title("GP sample 1")
+axes[2].imshow(Z[1, :].reshape(n_grid, n_grid), extent=[-10, 10, -10, 10])
+axes[2].set_title("GP sample 2")
```
```suggestion
_ = axes[2].set_title("GP sample 2")
```

awav commented on doc/source/notebooks/advanced/varying_noise.pct.py:
```diff
@@ -138,13 +152,20 @@ def objective_closure():
+    elinewidth=1.0,
+    color="C1",
+)
+plt.xlim(-5, 5)
```
```suggestion
_ = plt.xlim(-5, 5)
```

awav commented on doc/source/notebooks/advanced/varying_noise.pct.py:
```diff
@@ -225,12 +249,12 @@ def objective_closure():
-ax.plot(X, Y, 'C1x', mew=2)
-ax.set_xlim(-5, 5);
+ax.plot(X, Y, "C1x", mew=2)
+ax.set_xlim(-5, 5)
```
```suggestion
_ = ax.set_xlim(-5, 5)
```

awav commented on doc/source/notebooks/advanced/varying_noise.pct.py:
```diff
@@ -270,14 +297,15 @@ def objective_closure():
+ax.plot(X, Y, "C1x", mew=2)
 ax.set_xlim(-5, 5)
-ax.plot(xx, 2.5 * np.sin(6 * xx) + np.cos(3 * xx), 'C2--');
+ax.plot(xx, 2.5 * np.sin(6 * xx) + np.cos(3 * xx), "C2--")
```
```suggestion
_ = ax.plot(xx, 2.5 * np.sin(6 * xx) + np.cos(3 * xx), "C2--")
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -63,8 +64,8 @@
-plt.plot(x_grid, f_grid, 'C0', linewidth=1)
-plt.plot(x_grid, f_grid[:, 1], 'C0', linewidth=2);
+plt.plot(x_grid, f_grid, "C0", linewidth=1)
+plt.plot(x_grid, f_grid[:, 1], "C0", linewidth=2)
```
```suggestion
_ = plt.plot(x_grid, f_grid[:, 1], "C0", linewidth=2)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -75,9 +76,10 @@
-plt.plot(x_grid, p_grid, 'C1', linewidth=1)
-plt.plot(x_grid, p_grid[:, 1], 'C1', linewidth=2);
+plt.plot(x_grid, p_grid, "C1", linewidth=1)
+plt.plot(x_grid, p_grid[:, 1], "C1", linewidth=2)
```
```suggestion
_ = plt.plot(x_grid, p_grid[:, 1], "C1", linewidth=2)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -146,54 +151,61 @@ def objective():
-plt.plot(X[mask, 0], X[mask, 1], 'oC0', mew=0, alpha=0.5)
-plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], 'oC1', mew=0, alpha=0.5);
+plt.plot(X[mask, 0], X[mask, 1], "oC0", mew=0, alpha=0.5)
+plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], "oC1", mew=0, alpha=0.5)
```
```suggestion
_ = plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], "oC1", mew=0, alpha=0.5)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -208,11 +220,18 @@ def objective():
-plt.contour(xx, yy, p.numpy().reshape(*xx.shape), [0.5],  # plot the p=0.5 contour line only
-            colors='k', linewidths=1.8, zorder=100);
+plt.plot(X[mask, 0], X[mask, 1], "oC0", mew=0, alpha=0.5)
+plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], "oC1", mew=0, alpha=0.5)
```
```suggestion
_ = plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], "oC1", mew=0, alpha=0.5)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -208,11 +220,18 @@ def objective():
+plt.plot(X[mask, 0], X[mask, 1], "oC0", mew=0, alpha=0.5)
+plt.plot(X[np.logical_not(mask), 0], X[np.logical_not(mask), 1], "oC1", mew=0, alpha=0.5)
+
+plt.contour(
```
```suggestion
_ = plt.contour(
```

awav commented on doc/source/notebooks/basics/regression.pct.py:
```diff
@@ -38,18 +38,18 @@
 Y = data[:, 1].reshape(-1, 1)
 
-plt.plot(X, Y, 'kx', mew=2);
+plt.plot(X, Y, "kx", mew=2)
```
```suggestion
_ = plt.plot(X, Y, "kx", mew=2)
```

awav commented on doc/source/notebooks/basics/regression.pct.py:
```diff
@@ -165,15 +164,18 @@ def objective_closure():
+)
+
+plt.plot(xx, samples[:, :, 0].numpy().T, "C0", linewidth=0.5)
+plt.xlim(-0.1, 1.1)
```
```suggestion
_ = plt.xlim(-0.1, 1.1)
```

awav commented on doc/source/notebooks/tailor/mixture_density_network.pct.py:
```diff
@@ -28,28 +28,31 @@
-plt.ylabel("$y$");
+plt.plot(X, Y, "ro", alpha=0.3)
+plt.xlabel("$x$")
+plt.ylabel("$y$")
```
```suggestion
_ = plt.ylabel("$y$")
```

awav commented on doc/source/notebooks/theory/Sanity_check.pct.py:
```diff
@@ -46,31 +46,51 @@
+X = np.random.rand(20, 1) * 10
+Y = np.sin(X) + 0.9 * np.cos(X * 1.6) + np.random.randn(*X.shape) * 0.4
+Xtest = np.random.rand(10, 1) * 10
+plt.plot(X, Y, "kx", mew=2)
```
```suggestion
_ = plt.plot(X, Y, "kx", mew=2)
```

awav commented on doc/source/notebooks/theory/upper_bound.pct.py:
```diff
@@ -141,7 +147,7 @@ def plot_model(m, name=""):
 plt.xlabel("Number of inducing points")
 plt.ylabel("LML estimate")
-plt.legend();
+plt.legend()
```
```suggestion
_ = plt.legend()
```

awav commented on doc/source/notebooks/understanding/models.pct.py:
```diff
@@ -260,13 +273,13 @@ def closure():
-    plt.contour(xx, yy, p_test[:,i].reshape(200,200), [0.5], colors='k', linewidths=1)
-plt.scatter(X[:,0], X[:,1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis);
+    plt.contour(xx, yy, p_test[:, i].reshape(200, 200), [0.5], colors="k", linewidths=1)
+plt.scatter(X[:, 0], X[:, 1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis)
```
```suggestion
_ = plt.scatter(X[:, 0], X[:, 1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis)
```

awav commented on doc/source/notebooks/tailor/mixture_density_network.pct.py:
```diff
@@ -218,9 +221,9 @@ def moon_data(N, noise):
-plt.ylabel("$y$");
+plt.plot(X, Y, "ro", alpha=0.3)
+plt.xlabel("$x$")
+plt.ylabel("$y$")
```
```suggestion
_ = plt.ylabel("$y$")
```

awav commented on doc/source/notebooks/tailor/external-mean-function.pct.py:
```diff
@@ -64,7 +64,7 @@ def generate_data(num_functions=10, N=1000):
 
 # %%
-plt.plot(Xs, F);
+plt.plot(Xs, F)
```
```suggestion
_ = plt.plot(Xs, F)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -92,9 +94,9 @@ def invlink(f):
-plt.plot(X_gen, Y_gen, 'C3x', ms=8, mew=2);
+plt.plot(x_grid, p_grid[:, 1], "C1", linewidth=2)
+plt.plot(X_gen, p, "C1o", ms=6)
+plt.plot(X_gen, Y_gen, "C3x", ms=8, mew=2)
```
```suggestion
_ = plt.plot(X_gen, Y_gen, "C3x", ms=8, mew=2)
```

awav commented on doc/source/notebooks/basics/classification.pct.py:
```diff
@@ -39,11 +40,11 @@
 
 plt.figure(figsize=(10, 6))
-plt.plot(X, Y, 'C3x', ms=8, mew=2);
+plt.plot(X, Y, "C3x", ms=8, mew=2)
```
```suggestion
_ = plt.plot(X, Y, "C3x", ms=8, mew=2)
```

awav commented on doc/source/notebooks/advanced/kernels.pct.py:
```diff
@@ -120,15 +124,15 @@ def plotkernelsample(k, ax, xmin=-3, xmax=3):
 # plotting
 plt.figure()
-plt.plot(X2, K21);
+plt.plot(X2, K21)
```
```suggestion
_ = plt.plot(X2, K21)
```

awav commented on doc/source/notebooks/advanced/kernels.pct.py:
```diff
@@ -57,43 +59,44 @@
+plotkernelsample(gpflow.kernels.Linear(), axes[1, 1])
+plotkernelsample(gpflow.kernels.Cosine(), axes[1, 2])
+plotkernelsample(gpflow.kernels.Periodic(gpflow.kernels.SquaredExponential()), axes[1, 3])
+axes[0, 0].set_ylim(-3, 3)
```
```suggestion
_ = axes[0, 0].set_ylim(-3, 3)
```

awav commented on doc/source/notebooks/advanced/gps_for_big_data.pct.py:
```diff
@@ -214,8 +220,8 @@ def run_adam(model, iterations):
-plt.xlabel('iteration')
-plt.ylabel('ELBO');
+plt.xlabel("iteration")
+plt.ylabel("ELBO")
```
```suggestion
_ = plt.ylabel("ELBO")
```

awav commented on doc/source/notebooks/advanced/gps_for_big_data.pct.py:
```diff
@@ -56,10 +58,10 @@ def func(x):
 Xt = np.linspace(-1.1, 1.1, 1000)[:, None]
 Yt = func(Xt)
-plt.plot(Xt, Yt, c='k');
+plt.plot(Xt, Yt, c="k")
```
```suggestion
_ = plt.plot(Xt, Yt, c="k")
```

awav commented on doc/source/notebooks/advanced/coregionalisation.pct.py:
```diff
@@ -128,40 +132,45 @@ def objective_closure():
-print('B =', B)
-plt.imshow(B);
+print("B =", B)
+plt.imshow(B)
```
```suggestion
_ = plt.imshow(B)
```

awav commented on doc/source/notebooks/advanced/coregionalisation.pct.py:
```diff
@@ -61,12 +62,12 @@
-plt.plot(X1, Y1, 'x', mew=2)
-plt.plot(X2, Y2, 'x', mew=2);
+plt.plot(X1, Y1, "x", mew=2)
+plt.plot(X2, Y2, "x", mew=2)
```
```suggestion
_  = plt.plot(X2, Y2, "x", mew=2)
```

awav commented on doc/source/notebooks/theory/upper_bound.pct.py:
```diff
@@ -96,7 +101,7 @@ def plot_model(m, name=""):
 plt.xlabel("Number of inducing points")
 plt.ylabel("LML estimate")
-plt.legend();
+plt.legend()
```
```suggestion
_ = plt.legend()
```

awav commented on doc/source/notebooks/understanding/models.pct.py:
```diff
@@ -202,55 +205,65 @@ def closure():
-plt.scatter(X[:,0], X[:,1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis);
+
+matplotlib.rcParams["figure.figsize"] = (12, 6)
+plt.scatter(X[:, 0], X[:, 1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis)
```
```suggestion
_ = plt.scatter(X[:, 0], X[:, 1], 100, np.argmax(Y, 1), lw=2, cmap=plt.cm.viridis)
```

awav commented:
> Black strips all the trailing `;` that we used in code cells. The reason we had those was that pyplot commands often return axes or other plotting objects, and notebook cells print the return value of the last executed line - so this litters up the notebooks. I don't think we should merge this without discussing what to do about all those cases.

Fixed the issue with `_ =`. It was a mixture of `;` and `_ =`. Btw, `;` it bad syntax, really easy to make an error with it.

Added the `setup.py` as suggested.

st-- commented on doc/source/notebooks/advanced/mcmc.pct.py:
```diff
@@ -380,77 +395,74 @@ def run_chain_fn():
-plt.ylabel('hyper-parameter value');
+plt.legend(bbox_to_anchor=(1.0, 1.0))
+plt.xlabel("hmc iteration")
+plt.ylabel("hyper-parameter value")
```
```suggestion
_ = plt.ylabel("hyper-parameter value")
```

st-- commented on doc/source/notebooks/advanced/mcmc.pct.py:
```diff
@@ -474,13 +487,13 @@ def run_adam(model, iterations):
-plt.ylabel('neg_log_marginal_likelihood');
-
+plt.xlabel("iteration")
+plt.ylabel("neg_log_marginal_likelihood")
```
```suggestion
_ = plt.ylabel("neg_log_marginal_likelihood")
```

st-- commented on doc/source/notebooks/advanced/mcmc.pct.py:
```diff
@@ -535,44 +545,47 @@ def run_chain_fn():
-plt.plot(X, Y, 'kx', mew=2)
-plt.ylim(-0.1, np.max(np.percentile(rate_samples, 95, axis=0)));
+plt.plot(X, Y, "kx", mew=2)
+plt.ylim(-0.1, np.max(np.percentile(rate_samples, 95, axis=0)))
```
```suggestion
_ = plt.ylim(-0.1, np.max(np.percentile(rate_samples, 95, axis=0)))
```

st-- commented on doc/source/notebooks/advanced/mcmc.pct.py:
```diff
@@ -535,44 +545,47 @@ def run_chain_fn():
-plt.ylabel('hyper-parameter value');
+plt.legend(bbox_to_anchor=(1.0, 1.0))
+plt.xlabel("hmc iteration")
+plt.ylabel("hyper-parameter value")
```
```suggestion
_ = plt.ylabel("hyper-parameter value")
```
